x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/TransportDownsampleActionTest.java

```diff
@@ -0,0 +1,105 @@
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.List;
+import java.util.UUID;
+
+public class TransportDownsampleActionTest extends ESTestCase {
+    public void testCopyIndexMetadata() {
+        // GIVEN
+        final List<String> tiers = List.of(DataTier.DATA_HOT, DataTier.DATA_WARM, DataTier.DATA_COLD, DataTier.DATA_CONTENT);
+        final IndexMetadata source = new IndexMetadata.Builder(randomAlphaOfLength(10)).settings(
+            Settings.builder()
+                .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, randomIntBetween(1, 5))
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, randomIntBetween(1, 3))
+                .put(IndexMetadata.INDEX_DOWNSAMPLE_SOURCE_UUID_KEY, UUID.randomUUID().toString())
+                .put(IndexSettings.DEFAULT_PIPELINE.getKey(), randomAlphaOfLength(15))
+                .put(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey(), randomBoolean())
+                .put(DataTier.TIER_PREFERENCE_SETTING.getKey(), randomFrom(tiers))
+                .put(IndexMetadata.SETTING_INDEX_PROVIDED_NAME, randomAlphaOfLength(20))
+                .put(IndexMetadata.SETTING_CREATION_DATE, System.currentTimeMillis())
+        ).build();
+        final IndexMetadata target = new IndexMetadata.Builder(randomAlphaOfLength(10)).settings(
+            Settings.builder()
+                .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 6)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 4)
+                .put(IndexMetadata.INDEX_DOWNSAMPLE_SOURCE_UUID_KEY, UUID.randomUUID().toString())
+                .put(IndexSettings.DEFAULT_PIPELINE.getKey(), randomAlphaOfLength(15))
+                .put(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey(), randomBoolean())
+                .put(DataTier.TIER_PREFERENCE_SETTING.getKey(), randomFrom(tiers))
+                .put(IndexMetadata.SETTING_CREATION_DATE, System.currentTimeMillis())
+        ).build();
+        final IndexScopedSettings indexScopedSettings = new IndexScopedSettings(
+            Settings.EMPTY,
+            IndexScopedSettings.BUILT_IN_INDEX_SETTINGS
+        );
+
+        // WHEN
+        final Settings indexMetadata = TransportDownsampleAction.copyIndexMetadata(source, target, indexScopedSettings)
+            .build()
+            .getSettings();
+
+        // THEN
+        assertTargetSettings(target, indexMetadata);
+        assertSourceSettings(source, indexMetadata);
+        assertEquals(IndexMetadata.INDEX_UUID_NA_VALUE, indexMetadata.get(IndexMetadata.INDEX_DOWNSAMPLE_SOURCE_UUID_KEY));
+    }
+
+    private static void assertSourceSettings(final IndexMetadata indexMetadata, final Settings settings) {
+        assertEquals(indexMetadata.getIndex().getName(), settings.get(IndexMetadata.INDEX_DOWNSAMPLE_SOURCE_NAME_KEY));
+        assertEquals(
```

Comment:
Although it is possible to setup an index template for `downsample-` index prefix, I don't think this will happen often?
And I see this is an anti-pattern. My thinking here is that all the templates are defined on the data stream level. Whenever a new backing index of a data stream is created, the mappings/settings from the template that matches with the data stream is applied. I don't think that during the lifecycle of a backing index whenever we downsample, shrink or do create a searchable snapshot, we should apply a whole different template. Even though the concrete backing index gets replaced by another concrete backing index.

I don't think we should let custom `_tier_preference` get applied here. ILM is actually in charge of that. I think we should maybe even prevent templates from being applied the backing indices getting downsampled.

Reply:
++ I think we shouldn't worry about `_tier_preference` being custom configured in index templates and we should copy it from the source index. As Martijn said, ILM is in charge of transitioning the indices through tiers and will change this setting as part of the `migrate` action in different phases. 

Custom allocations will still be able to exist via the custom node attributes.